File: 02-12-a3-data/ex.py
import re
from matplotlib import pylab

name_p = r">([A-Z- ]+)<"
money_p = r">\$([0-9.,]+)<"

salaries = []

f = open("data/output.txt", "r")
for line in f.readlines():
    m1 = re.findall(name_p, line)
    if m1: print m1
    m2 = re.findall(money_p, line)
    if m2: 
        print m2[0]
        s = m2[0]
        amt = float(s.replace(",", ""))
        if amt > 100000: salaries.append(amt)

pylab.hist(salaries, 100)
pylab.show()

File: 02-12-a3-data/ex.java
import java.util.regex.*;
import java.io.*;

class ex {
    public static void main(String[] args) throws Exception {
        BufferedReader reader = new BufferedReader(
                new FileReader("data/output.txt"));
        String line = reader.readLine();
        Pattern namep = Pattern.compile(">([A-Z- ]+)<");
        Pattern moneyp = Pattern.compile(">\\$([0-9,.]+)<");
        while(line != null) {
            Matcher m1 = namep.matcher(line);
            while(m1.find())
                System.out.println(m1.group(1));

            Matcher m2 = moneyp.matcher(line);
            while(m2.find())
                System.out.println(m2.group(1));
            line = reader.readLine();
        }
    }
}
